app/elements/batch-image-search/batch-image-search.html

```diff
@@ -0,0 +1,379 @@
+        imageType: {
+          type: String,
+          value: ''
+        },
+        //Store all Images
+        images: {
+          type: Array,
+          value: []
+        },
+        // Store all sha1 of the images result
+        sha1Images: {
+          type: Array,
+          value: []
+        },
+        // Count of rendered results
+        renderedCount: {
+          type: Number,
+          value: 0
+        },
+        // enable or disable search button
+        boolButton: {
+          type: Boolean,
+          notify: true,
+          value: true
+        },
+        // temp to store source image
+        compareSourceImage: {
+          type: String,
+          notify: true,
+          value: ''
+        },
+
+        /**
+         * Whether the data is loading.
+        */
+        loading: {
+          type: Boolean,
+          value: false,
+          notify: true
+        },
+        // upload input
+        _base64Input: {
+          type: String,
+          value: ''
+        },
+        // Image input
+        _imageInput: {
+          type: String,
+          value: ''
+        },
+        // Mimetype of the uploaded image
+        mimeType: {
+          type: String,
+          value: ''
+        },
+        // Blur class
+        blur: {
+          type: Boolean,
+          notify: true
+        },
+
+        observers: [
+          'userActivity(value)'
+        ],
+      },
+
+      displayLink: function(index) {
+        var shaId = this.sha1Images[index];
+        var urlResp = '/image.html?value=http://dig.isi.edu/ht/data/' + shaId + '&field=_id';
+        return urlResp;
+      },
+
+      getBlurStyleClass: function(blur) {
+        return (blur ? 'small-blur' : '');
+      },
+
+      userActivity: function() {
+        if(this._imageInput) {
+          this.boolButton = false;
+          this.imageType = 'url';
+        } else {
+          this.boolButton = true;
+          this.imageType = '';
+        }
+      },
+
+      handleResponse: function(event) {
+        var resp = event.detail.response;
+        this.images = resp.images[0].similar_images.cached_image_urls;
+        this.sha1Images = resp.images[0].similar_images.sha1;
+        this.$.resultImages.open();
+        this.clearInput();
+        this.$$('#batchImageSearchDialog').close();
+      },
+
+      makeHeaders: function(user, password) {
+        return 'Basic ' + btoa(user + ':' + password);
+      },
+
+      _createSearchListener: function() {
+        var self = this;
+        return {
+          onClick: function() {
+            self.$.xhr.params = {};
+            if(self._imageInput || self._base64Input) {
+              if(self.imageType === 'url') {
+                self.$.xhr.params.url = self._imageInput;
+                self.compareSourceImage = self._imageInput;
+                self.$.xhr.method = 'GET';
+                self.$.xhr.url = 'https://isi.memexproxy.com/ColumbiaUimgSearch.php';
```

Comment:
The two URLs for the image service (here and line 335) should be configuration properties (in the local.env.js file).  We don't want to have to repackage the app if the service changes URLs.